database/influxdb/influxdb_database.py

```diff
@@ -0,0 +1,180 @@
+        default: root
+    port:
+        description:
+            - The port on which InfluxDB server is listening
+        default: 8086
+    database_name:
+        description:
+            - Name of the database that will be created/destroyed
+        required: True
+    if_not_exists:
+        description:
+            - Determines if database should be created only if it does not exist
+        default: False
+    state:
+        description:
+            - Determines if the database should be created or destroyed
+        choices: ['present', 'absent']
+        default: present
+'''
+
+EXAMPLES = '''
+# Example influxdb_database command from Ansible Playbooks
+- name: Create database
+    influxdb_database:
+      hostname: "{{influxdb_ip_address}}"
+      database_name: "{{influxdb_database_name}}"
+      state: present
+
+- name: Destroy database
+    influxdb_database:
+      hostname: "{{influxdb_ip_address}}"
+      database_name: "{{influxdb_database_name}}"
+      state: absent
+
+- name: Create database using custom credentials
+    influxdb_database:
+      hostname: "{{influxdb_ip_address}}"
+      username: "{{influxdb_username}}"
+      password: "{{influxdb_password}}"
+      database_name: "{{influxdb_database_name}}"
+      state: present
+'''
+
+RETURN = '''
+#
+'''
+
+import requests.exceptions
+try:
+    from influxdb import InfluxDBClient
+    from influxdb import exceptions
+    HAS_INFLUXDB = True
+except ImportError:
+    HAS_INFLUXDB = False
+
+
+def influxdb_argument_spec():
+    return dict(
+        hostname=dict(required=True),
+        port=dict(default=8086, type='int'),
+        username=dict(default='root'),
+        password=dict(default='root', no_log=True),
+        database_name=dict(required=True)
+    )
+
+
+def connect_to_influxdb(module):
+    hostname = module.params['hostname']
+    port = module.params['port']
+    username = module.params['username']
+    password = module.params['password']
+    database_name = module.params['database_name']
+
+    client = InfluxDBClient(
+        host=hostname,
+        port=port,
+        username=username,
+        password=password,
+        database=database_name
+    )
+    return client
+
+
+def create_database(module):
+    database_name = module.params['database_name']
+    if_not_exists = module.params['if_not_exists']
+    client = connect_to_influxdb(module)
+
+    if not module.check_mode:
+        try:
+            client.create_database(database_name, if_not_exists=if_not_exists)
+        except requests.exceptions.ConnectionError:
+            e = get_exception()
+            module.fail_json(msg=str(e))
+
+    module.exit_json(changed=True)
+
+
+def drop_database(module):
+    database_name = module.params['database_name']
+    client = connect_to_influxdb(module)
+
+    if not module.check_mode:
+        try:
+            client.drop_database(database_name)
+        except exceptions.InfluxDBClientError:
+            e = get_exception()
```

Comment:
Note: there's nothing wrong with using e = get_exception() here but it's also okay to use ```except exceptions.InfluxDBClientError as e:``` when the module needs python >=2.6 for another reason (I'm assuming that it needs it for the influxdb python moudle to function).